inference-engine/src/mkldnn_plugin/utils/multidim_map.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <unordered_map>
+#include <type_traits>
+#include <functional>
+
+namespace MKLDNNPlugin {
+
+namespace internal {
+
+template <typename K>
+struct enum_hash {
+    std::size_t operator()(K t) const {
+        return static_cast<std::size_t>(t);
+    }
+};
+
+template<typename K>
+using hash_t = typename std::conditional<std::is_enum<K>::value, enum_hash<K>, std::hash<K>>::type;
+
+}   // namespace internal
+
+template<typename K, typename ...Ts>
+struct multidim_map {
```

Comment:
do we checked that it is not already implemented somewhere in IE or other parts of the project?